keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/unary/SumVertexTest.java

```diff
@@ -28,7 +31,46 @@ public void doesSum() {
     }
 
     @Test
-    public void doesSumSimpleAutoDiff() {
+    public void doesSumAllSpecifiedDimensions() {
+        DoubleVertex a = new UniformVertex(new int[]{1, 5}, 0, 10);
+        a.setValue(new double[]{1, 2, 3, 4, 5});
+
+        DoubleVertex summed = a.sum(0, 1);
+        DoubleTensor expected = DoubleTensor.scalar(1 + 2 + 3 + 4 + 5);
+
+        assertEquals(expected, summed.eval());
+    }
+
+    @Test
+    public void doesSumSpecifiedDimensions() {
+        int[] shape = {2, 2, 2, 2};
+        DoubleVertex a = new UniformVertex(shape, 0, 10);
+        a.setValue(DoubleTensor.arange(0, TensorShape.getLength(shape)).reshape(shape));
+
+        DoubleVertex summed = a.sum(0, 2);
+
+        DoubleTensor expected = DoubleTensor.create(new double[]{20, 24, 36, 40}, 2, 2);
+
+        assertEquals(expected, summed.eval());
+    }
+
+    @Test
+    public void doesCalculateCorrectShape() {
+        int[] shape = {2, 3, 4, 5, 6, 1};
+        DoubleVertex a = new UniformVertex(shape, 0, 10);
+        DoubleTensor highrank = DoubleTensor.arange(0, TensorShape.getLength(shape)).reshape(shape);
+        a.setValue(highrank);
+
+        assertArrayEquals(new int[]{3, 5, 6, 1}, a.sum(0, 2).getShape());
+        assertArrayEquals(new int[]{5, 6}, a.sum(0, 1, 2, 5).getShape());
+        assertArrayEquals(new int[]{6, 1}, a.sum(0, 1, 2, 3).getShape());
+        assertArrayEquals(new int[]{2, 1}, a.sum(1, 2, 3, 4, 5).getShape());
+        assertArrayEquals(new int[]{1, 3}, a.sum(0, 2, 3, 4, 5).getShape());
+        assertArrayEquals(new int[]{1, 1}, a.sum(0, 1, 2, 3, 4, 5).getShape());
```

Comment:
Should we have a test here for things that start and result in a scalar?

Reply:
👍